kf_lib_data_ingest/etl/configuration/target_api_config.py

```diff
@@ -227,52 +234,98 @@ def _validate_target_concept_attr_mappings(self):
+                self._validate_link_list(target_concept_dict)
+
+    def _validate_properties(self, target_concept, target_concept_dict):
+        """
+        Validate properties dict for each target concept config dict
+
+        Keys must be strs
+        Values must be existing standard concept strings
+        """
+        props_dict = target_concept_dict['properties']
+
+        # All property keys are strs
+        assert_all_safe_type(props_dict.keys(), str)
+
+        # Are all mapped values valid standard concept attributes?
+        for target_attr, mapped_attr in props_dict.items():
+            if mapped_attr is None:
+                continue
+
+            mapped_attr = str(mapped_attr)
+            if mapped_attr not in concept_property_set:
+                raise ValueError(
+                    f'Error in dict for {target_concept} '
+                    'All target concept attributes must be mapped to '
+                    'an existing standard concept attribute. Mapped '
+                    f'attribute {mapped_attr} for target attr '
+                    f'{target_concept}.{target_attr} does not exist')
+
+    def _validate_link_list(self, target_concept_dict):
+        """
+        Validate the 'links' element in a target concept's config dict
+
+        'links' must be a list of dicts
+        Each dict must have the required keys:
+            - target_attribute
+            - standard_concept
+            - target_concept
+        'target_attribute' must be a string
+        'standard_concept' must point to an existing standard concept
+        'target_concept' must point to an existing target_concept
+        """
+        valid_target_concepts = set(self.contents.target_concepts.keys())
+        links = target_concept_dict['links']
+        assert_safe_type(links, list)
+        assert_all_safe_type(links, dict)
+
+        for link_dict in links:
+            # Check link dict keys
+            required_link_keys = {'target_attribute',
+                                  'standard_concept',
+                                  'target_concept'}
+            if not all([k in required_link_keys for k in link_dict]):
+                raise KeyError(
+                    f'Badly formatted link dict in \n'
+                    f'{target_concept_dict}. Missing required keys: '
+                    f'{required_link_keys}'
+                )
+            # Check target_attribute
+            assert_safe_type(link_dict['target_attribute'], str)
+            # Check standard concept
+            sc = link_dict['standard_concept']
+            if sc not in concept_set:
+                sc_name = sc
+                try:
+                    sc_name = sc._CONCEPT_NAME
+                except AttributeError:
+                    pass
+                raise ValueError(
+                    'Invalid link found in target concept config \n'
+                    f'{pformat(target_concept_dict)}! '
```

Comment:
It's probably not a good idea to put punctuation adjacent to a data structure.

```suggestion
                    f'{pformat(target_concept_dict)}'
```